table/block_based_table_reader.cc

```diff
@@ -2406,6 +2407,28 @@ Status BlockBasedTable::Get(const ReadOptions& read_options, const Slice& key,
           break;
         }
 
+        bool hash_effective, found;
+        biter.SeekForGet(key, &hash_effective, &found);
+
+        if (hash_effective) {
+          if (found) {
+            ParsedInternalKey parsed_key;
+            s = Status::OK();
+
+            if (!ParseInternalKey(biter.key(), &parsed_key)) {
+              s = Status::Corruption(Slice());
```

Comment:
1. Should you be setting a valid Corruption message here?
2. I believe you also need to break here without proceeding to SaveValue on the next line.